src/model/WaterHeaterMixed.cpp

```diff
@@ -1216,6 +1220,123 @@ namespace model {
       }
     }
 
+    boost::optional<PlantLoop> WaterHeaterMixed_Impl::sourceSidePlantLoop() const {
+      if (m_sourceSidePlantLoop) {
+        return m_sourceSidePlantLoop;
+      } else {
+        boost::optional<HVACComponent> sourceSideOutletHVACComponent;
+
+        if (auto t_sourceSideOutletModelObject = sourceSideOutletModelObject()) {
+          sourceSideOutletHVACComponent = t_sourceSideOutletModelObject->optionalCast<HVACComponent>();
+        }
+
+        if (sourceSideOutletHVACComponent) {
+          auto plantLoops = this->model().getConcreteModelObjects<PlantLoop>();
+
+          for (const auto& plantLoop : plantLoops) {
+            if (!plantLoop.supplyComponents(plantLoop.supplyInletNode(), sourceSideOutletHVACComponent.get()).empty()) {
+              m_sourceSidePlantLoop = plantLoop;
+              return plantLoop;
+            }
+          }
+        }
+      }
+
+      return boost::none;
+    }
+
+    boost::optional<PlantLoop> WaterHeaterMixed_Impl::useSidePlantLoop() const {
+      return plantLoop();
+    }
+
+    bool WaterHeaterMixed_Impl::removeFromSourceSidePlantLoop() {
+      if (auto plant = sourceSidePlantLoop()) {
+        m_sourceSidePlantLoop = boost::none;
+        return HVACComponent_Impl::removeFromLoop(plant->supplyInletNode(), plant->supplyOutletNode(), demandInletPort(), demandOutletPort());
+      }
+
+      return false;
+    }
```

Comment:
IMHO, this should just be a name alias to removeFromSecondaryPlantLoop.

Override removeFromSecondaryPlantLoop (and make it virtual in the WaterToWaterComponent_Impl class).
And make it disconnect supply and demand side alike, but only for the source side.